File: Visual Studio/Applications/Bouncing Ball/Bouncing Ball/utilities.c
#include "utilities.h"

int RoundToInt(double x)
{
  int s = (int)x;
  double t = fabs(x - s);

  if ((t < 0.5) || (t == 0.5 && s % 2 == 0))
  {
    return s;
  }
  else
  {
    if (x < 0)
    {
      return s - 1;
    }
    else
    {
      return s + 1;
    }
  }
}

double Square(double x)
{
  return x * x;
}

File: Visual Studio/Applications/The Matrix/The Matrix/Utilities.cs
using System;

namespace TheMatrix
{
    internal static class Utilities
    {
        private static string base_string = "012345678ABCDEFGHIJKLMNOPQRabcdefghijklmnopqrstuvwxyz"; // "!\"#$%&'()*+,-./0123456789:;<=>?ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_abcdefghijklmnopqrstuvwxyz{|}~ ;";

        static Utilities()
        {
            Random = new Random();
        }

        public static Random Random
        {
            get;
            private set;
        }

        public static bool Probability(double p)
        {
            return Random.NextDouble() < p;
        }

        public static double GetRandomDouble(double min, double max)
        {
            return min + Random.NextDouble() * (max - min);
        }

        public static int RoundToInt(double x)
        {
            return (int)Math.Round(x);
        }

        public static char GetRandomCharacter()
        {
            return base_string[Random.Next(base_string.Length)];
        }
    }
}
